dll/win32/iphlpapi/iphlpapi_main.c

```diff
@@ -2620,12 +2622,93 @@ SetIpForwardEntryToStack(PMIB_IPFORWARDROW pRoute)
 }
 
 DWORD WINAPI
-NhGetInterfaceNameFromDeviceGuid(DWORD dwUnknown1,
-                                 DWORD dwUnknown2,
-                                 DWORD dwUnknown3,
+NhGetInterfaceNameFromDeviceGuid(GUID * pInterfaceGUID,
+                                 PWCHAR pInterfaceName,
+                                 PULONG pOutBufLen,
                                  DWORD dwUnknown4,
                                  DWORD dwUnknown5)
 {
-    FIXME("NhGetInterfaceNameFromDeviceGuid() stub\n");
-    return 0L;
+    LPOLESTR pStr;
+    WCHAR szKeyName[2*MAX_PATH];
+    HKEY hKey;
+    DWORD cbData = 0, type = REG_SZ;
+
+    if (StringFromCLSID(pInterfaceGUID, &pStr) != ERROR_SUCCESS)
```

Comment:
I'd go for `StringFromGUID2`, that saves having to free it, but in this case it adds an extra import to `ole32`, so maybe use `RtlStringFromGUID`. (Altho that needs a free again),
or roll your own:
https://git.reactos.org/?p=reactos.git;a=blob;f=modules/rostests/apitests/setupapi/devclass.c;hb=68af472a2133fb8e6859a7a9d5363385b995f891#l42
https://git.reactos.org/?p=reactos.git;a=blob;f=dll/win32/setupapi/devinst.c;hb=68af472a2133fb8e6859a7a9d5363385b995f891#l79